src/pybind/mgr/mgr_module.py

```diff
@@ -343,6 +343,67 @@ def _unit_to_str(self, unit):
         elif unit == self.BYTES:
             return "B/s"  
     
+    def to_pretty_iec(self, n):
+        bump_after = 10
+        if n > bump_after << 60:
+            return str(n >> 60) + ' Ei'
+        if n > bump_after << 50:
+            return str(n >> 50) + ' Pi'
+        if n > bump_after << 40:
+            return str(n >> 40) + ' Ti'
+        if n > bump_after << 30:
+            return str(n >> 30) + ' Gi'
+        if n > bump_after << 20:
+            return str(n >> 20) + ' Mi'
+        if n > bump_after << 10:
+            return str(n >> 10) + ' Ki'
+        return str(n) + ' '
+
+    def get_pretty_row(self, elems, width):
+        """
+        Takes an array of elements and returns a string with those elements
+        formatted as a table row. Useful for polling modules.
+
+        :param elems: the elements to be printed
+        :param width: the width of the terminal
+        """
+        n = len(elems)
+        column_width = width / n
+
+        ret = '|'
+        for elem in elems:
+            ret += '{0:>{w}} |'.format(elem, w=column_width - 2)
+
+        return ret
+
+    def get_pretty_header(self, elems, width):
+        """
+        Like ``get_pretty_row`` but adds dashes, to be used as a table title.
+
+        :param elems: the elements to be printed
+        :param width: the width of the terminal
+        """
+        n = len(elems)
+        column_width = width / n
+
+        # dash line
+        ret = '+'
```

Comment:
could you leverage python-prettytable to do the formatting?

Reply:
@tchaikov I had explored prettytable initially (especially since it's already used in the daemonperf), but as @jcsp said, it wasn't a straightforward match with this use case since we're not really "printing a table" here.